hello_world/app.py

```diff
@@ -1,42 +1,57 @@
 import json
+import os
+import uuid
+import boto3
 
-# import requests
+
+# Initialize the DynamoDB client
+dynamodb = boto3.resource('dynamodb')
+table = dynamodb.Table(os.environ.get("TABLE_NAME"))
 
 
 def lambda_handler(event, context):
-    """Sample pure Lambda function
+   try:
+       # Get the long URL from the POST request body
+       body = json.loads(event.get("body", "{}"))
+       long_url = body.get("long_url")
+
+
+       if not long_url:
+           return {
+               "statusCode": 400,
+               "body": json.dumps({"error": "long_url is a required field"}),
+           }
 
-    Parameters
-    ----------
-    event: dict, required
-        API Gateway Lambda Proxy Input Format
 
-        Event doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+       # Generate a unique short ID (I will replace this with base62 later)
+       short_id = str(uuid.uuid4())[:7]
 
-    context: object, required
-        Lambda Context runtime methods and attributes
 
-        Context doc: https://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html
+       # Save the mapping to DynamoDB
+       table.put_item(
+           Item={
+               'id': short_id,
+               'long_url': long_url
+           }
+       )
 
-    Returns
-    ------
-    API Gateway Lambda Proxy Output Format: dict
 
-        Return doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html
-    """
+       # Construct the short URL to return to the user
+       api_gateway_url = f"https://{event['requestContext']['domainName']}/{event['requestContext']['stage']}"
+       short_url = f"{api_gateway_url}/{short_id}"
```

Comment:
I don't understand this right now. I think you are trying to extract our domain name and prepend. Like if the id is `123456` then you need `tinyurl.com/123456` to return to customer and you are trying to get the `tinyurl.com` from the context of the API request. But you can also make it hardcoded maybe. Of course it won't work but that is fine for now. The reason is this will create a even bigger url :).

just pick a domain name for your website-to-be `www.tariksurls.com` then you can return `f"tariks.com/{short_id}` to the customer.